modules/gcp/centos-std/centos-std-startup.sh.tmpl

```diff
@@ -92,6 +117,35 @@ install_pcoip_agent() {
     fi
 }
 
+install_idle_shutdown() {
+    log "--> Installing idle shutdown..."
+    mkdir /tmp/idleShutdown
+
+    retry wget "https://raw.githubusercontent.com/teradici/deploy/master/remote-workstations/new-agent-vm/Install-Idle-Shutdown.sh" -O /tmp/idleShutdown/Install-Idle-Shutdown-raw.sh
+    
+    awk '{ sub("\r$", ""); print }' /tmp/idleShutdown/Install-Idle-Shutdown-raw.sh > /tmp/idleShutdown/Install-Idle-Shutdown.sh && chmod +x /tmp/idleShutdown/Install-Idle-Shutdown.sh
+
+    log "--> Setting auto shutdown idle timer to $${AUTO_SHUTDOWN_IDLE_TIMER} minutes"
+    INSTALL_OPTS="--idle-timer $${AUTO_SHUTDOWN_IDLE_TIMER}"
+    if [[ "$${ENABLE_AUTO_SHUTDOWN}" = "false" ]]; then
+        INSTALL_OPTS="$${INSTALL_OPTS} --disabled"
+    fi
+    
+    retry /tmp/idleShutdown/Install-Idle-Shutdown.sh $${INSTALL_OPTS}
+
+    exitCode=$?
+    if [[ $exitCode -ne 0 ]]; then
+        log "--> Failed to install idle shutdown."
+        exit 1
+    fi
+
+    if [[ $CPU_POLLING_INTERVAL -ne 15 ]]; then
+        log "--> Setting CPU polling interval to $${CPU_POLLING_INTERVAL} minutes"
+        sed -i "s/OnUnitActiveSec=15min/OnUnitActiveSec=$${CPU_POLLING_INTERVAL}min/g" /etc/systemd/system/CAMIdleShutdown.timer.d/CAMIdleShutdown.conf
```

Comment:
The way to set CPU polling for Linux is different than Windows, I consulted Charanpreet on this. When changed to 1 min, the idle timer is much shorter and works as expected (instead of 15 + 15 mins for example).